src/Sylius/Bundle/OrderBundle/Modifier/AdjustmentManager.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * @author    Pete Ward <peter.ward@reiss.com>
+ * @date      05/02/2016
+ * @copyright Copyright (c) Reiss Clothing Ltd.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\OrderBundle\Modifier;
+
+use Doctrine\ORM\EntityManager;
+use Sylius\Component\Order\Model\Adjustment;
+use Sylius\Component\Order\Factory\AdjustmentFactory;
+use Sylius\Component\Order\Model\Order;
+use Sylius\Component\Order\Model\OrderItemUnit;
+
+/**
+ * @author Pete Ward <peter.ward@reiss.com>
+ */
+class AdjustmentManager
+{
+    /**
+     * @var AdjustmentFactory
+     */
+    private $adjustmentFactory;
+
+    /**
+     * @var EntityManager
+     */
+    private $entityManager;
+
+    /**
+     * @param AdjustmentFactory $adjustmentFactory
+     * @param EntityManager     $entityManager
+     */
+    public function __construct(AdjustmentFactory $adjustmentFactory, EntityManager $entityManager)
+    {
+        $this->adjustmentFactory = $adjustmentFactory;
+        $this->entityManager     = $entityManager;
+    }
+
+    /**
+     * @param OrderItemUnit $unit
+     * @param string        $descriptionSuffix
+     */
+    public function buildRefundAdjustmentsForUnit(OrderItemUnit $unit, $descriptionSuffix = 'Return')
+    {
+        if (count($unit->getRefundAdjustments())) {
+            // If already has some, do nothing.
+            return;
+        }
+
+        foreach ($unit->getAdjustments() as $adjustment) {
+            $refundAdjustment = clone $adjustment;
```

Comment:
This could be done in factory, just maybe change the method:
```php
public function createUnitRefund(OrderItemUnit $unit, $label = 'Refund', Adjustment $baseAdjustment = null)
{
   $adjustment = null !== $baseAdjustment ? clone $baseAdjustment : $this->createNew();
   // ...
}